SalesforceFoundation/Cumulus
# Adding Setting for Selecting Allocation Type

# Critical Changes

# Changes

- Adding "Allocation Type" Picklist to the GAU Allocations Settings page, which allows selection of Opportunity or Payment.
- Updating the Manage Allocations page, accessible from the GAU Allocations related list, to work with the Payment Object.

# Issues Closed

# New Metadata

Custom Labels: 
- alloDefaultNotEnabled
- alloManagePaymentAllocations
- stgAllocationLevel
- stgLabelAllocationBehaviorSettings
- stgHelpAllocationLevel
- stgHelpAllocationLevelWarning

# Deleted Metadata

# Testing Notes

- Install NPSP, create a GAU.
- Go to NPSP Settings → Donations → GAU Allocations and set Allocation Type to “Payment” with Default Allocations Enabled unchecked.
- A warning message will be displayed after selecting "Payment" if any Opportunities have associated Allocations in this org.
- Save, and make sure the "You must enable Default Allocations" error message appears.
- With Allocation Type still set to “Payment”, check the box next to Default Allocations Enabled, then Save.
- Another warning should appear saying that a Default GAU is required.
- Finally, select a Default GAU, then save, which should work without any errors.
- While Payment Allocations are enabled: Modify the Payment page layout to include the GAU Allocations related list.
    - Also add the "Manage Allocations" button to that list.
    - This link has instructions for adding a button: https://success.salesforce.com/answers?id=9063A000000lFzwQAE
- Navigate to, or create, a Payment, then ensure that the Manage Allocations page works as it should.
